lib/shared/puppet_forge/unpacker.rb

```diff
@@ -0,0 +1,78 @@
+require 'pathname'
+require 'tmpdir'
+require 'shared/puppet_forge/error'
+require 'shared/puppet_forge/tar'
+
+module PuppetForge
+  class Unpacker
+    # Unpack a tar file into a specified directory
+    #
+    # @param filename [String] the file to unpack
+    # @param target [String] the target directory to unpack into
+    def self.unpack(filename, target, tmpdir)
+      inst = self.new(filename, target, tmpdir)
+      inst.unpack
+      inst.move_into(target)
+    end
+
+    # Set the owner/group of the target directory to those of the source
+    #
+    # @param source [Pathname] source of the permissions
+    # @param target [Pathname] target of the permissions change
+    # @param microsoft_windows boolean flag for whether this system is windows or not
+    def self.harmonize_ownership(source, target, microsoft_windows = false)
+      if !microsoft_windows
+        FileUtils.chown_R(source.stat.uid, source.stat.gid, target)
+      end
+    end
+
+    # @param filename [String] the file to unpack
+    # @param target [String] the target directory to unpack into
+    def initialize(filename, target, tmpdir)
+      @filename = Pathname.new(filename)
+      @target = target
+      @tmpdir = tmpdir
```

Comment:
The `#tmpdir` method memoizes the `@tmpdir` value, but here we assign it - so this assignment looks redundant.